kamelets/aws-s3-source.kamelet.yaml

```diff
@@ -107,13 +107,28 @@ spec:
         description: The number of milliseconds before the next poll of the selected bucket.
         type: integer
         default: 500
+      outputFormat:
```

Comment:
the list of possible  types must be expressed through an enum

Reply:
why limiting this to a predefined enum of formats? The user may provide a custom format, too.

Reply:
the problem is: how the user/tooling know what converters are available and what are supported ?

Reply:
its in the Kamelet specification. The default supported output/input types are specified and described in the Kamelet spec. Like this:

```yaml
apiVersion: camel.apache.org/v1alpha1
kind: Kamelet
metadata:
  name: aws-s3-source
  labels:
    camel.apache.org/kamelet.type: "source"
spec:
  definition:
    title: "AWS S3 Source"
    type: object
    properties:
      bucketNameOrArn:
        title: Bucket Name
        description: The S3 Bucket name or ARN
        type: string
      ...
  output:
    default: binary
    types:
      binary:
        mediaType: application/octet-stream
      json:
        mediaType: application/json
        dependencies:
          - "camel:jackson"
        schema:
          type: object
          properties:
            key:
              title: S3 key
              description: The S3 key identifier
              type: string
            fileContent:
              title: File content
              description: The file content as String
              type: string
```

This is the place where each data type is able to specify a schema and additional dependencies, too.

Reply:
yes but if we expect the user to provide custom converter, then  the only way to validate it is to wait for runtime failures. I'm thinking about how tooling would provide support/validation for that.